### ISSUE TITLE
Harden fcrepo

### ISSUE THREAD

**Issue by decimalator** [OregonDigital/OD2#1299]
Fedora ran into a problem ingesting new works, throwing 500 errors

```
["Ldp::HttpError","STATUS: 500 javax.jcr.RepositoryException: org.modeshape.persistence.file.FileProviderException: cannot write new value for the first time\n\tat org.modeshape.jcr.JcrSession.save(JcrSession.java:1196)\n\tat org.fcrepo.kernel.modeshape.FedoraSessionImpl.commit(FedoraSessionImpl.java:95)\n\tat org.fcrepo.http.commons.session.HttpSession.commit(HttpSession.java:55)\n\tat org.fcrepo.http.api.FedoraLdp.createObject(FedoraLdp.java:535)\n\tat sun.reflect.GeneratedMethodAccessor54.invoke(Unknown Source)\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n\tat java.lang.reflect.Method.invoke(Method.java:498)\n\tat org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81)\n\tat org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144)\n\tat org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(Abstrac..."]

process_action.action_controller.exception_object
STATUS: 500 javax.jcr.RepositoryException: org.modeshape.persistence.file.FileProviderException: cannot write new value for the first time at org.modeshape.jcr.JcrSession.save(JcrSession.java:1196) at org.fcrepo.kernel.modeshape.FedoraSessionImpl.commit(FedoraSessionImpl.java:95) at org.fcrepo.http.commons.session.HttpSession.commit(HttpSession.java:55) at org.fcrepo.http.api.FedoraLdp.createObject(FedoraLdp.java:535) at sun.reflect.GeneratedMethodAccessor54.invoke(Unknown Source) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81) at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144) at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(Abstrac...
```

Our Fedora deployment needs to be hardened and productionized, and some metrics endpoints added so we can see what it's doing.

**Comment by CGillen** [comment]
@decimalator Is this still an issue?

**Comment by decimalator** [comment]
This has been resolved.

I forget when exactly, but we switched from the filesystem Modeshape metadata repository to MySQL. That solved all of the Modeshape issues. 

Since then we migrated the MySQL database from the shared `lib-mdb2` instance to the `fcrepo-db` StatefulSet running in both the production and test namespaces. For persistent storage they use a Local disk that's backed by SSD storage on the node. We're now able to scale up over 15k SQL queries per second!

We also moved from the ualbertalib 4.7.3 containers we started with and built our own images using the latest 4.7.5.

We don't have ALL of the metrics I would like, but we at least can gather errors from logs in Loki, MySQL database metrics from a MySQL metrics exporter, etc. 

We can open new tickets for any new Fedora issues
